python/plugins/MetaSearch/dialogs/maindialog.py

```diff
@@ -748,8 +750,7 @@ def add_to_ows(self):
         self.settings.endGroup()
 
         # open provider window
-        ows_provider = QgsProviderRegistry.instance().createSelectionWidget(stype[2],
-                                                                            self)
+        ows_provider = QgsGui.providerGuiRegistry().sourceSelectProviders(stype[2])[0].createDataSourceWidget()
```

Comment:
```suggestion
        ows_provider = QgsGui.sourceSelectProviderRegistry().createSelectionWidget(stype[2], self, Qt.Widget,  QgsProviderRegistry.WidgetMode.Embedded)
```

Does that work? Should be a bit more resilient because it has checks to ensure that the requested provider exists...

Reply:
Yes it works. With both (setModal+show) and (setmodality+exec_). Should I revert the changes I made there ( I made them because the dialog was disappearing as soon as opened)?

Reply:
Yeah let's revert them. Who knows what side effects that could have..!